app/src/androidTest/java/com/github/epfl/meili/home/GoogleSignInActivityTest.kt

```diff
@@ -24,53 +21,68 @@ import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
 class GoogleSignInActivityTest {
+    private lateinit var mockService: MockAuthenticationService
+
     companion object {
         private const val MOCK_NAME = "Fake Name"
     }
 
     @get:Rule
     var testRule: ActivityScenarioRule<GoogleSignInActivity?>? = ActivityScenarioRule(
-            GoogleSignInActivity::class.java
+        GoogleSignInActivity::class.java
     )
 
     @Before
     fun before() {
         runOnUiThread {
-            AuthenticationService.signOut()
-            AuthenticationService.isLoggedIn.value = false
-            AuthenticationService.email = null
-            AuthenticationService.name = null
+            //Injecting authentication Service
+            mockService = MockAuthenticationService()
+            Auth.setAuthenticationService(mockService)
+
+            Auth.signOut()
+            Auth.isLoggedIn.value = false
+            Auth.email = null
+            Auth.name = null
         }
     }
 
-    private fun getGSO(): GoogleSignInOptions {
-        return GoogleSignInOptions
-                .Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
-                .requestIdToken(
-                        MainApplication.applicationContext().getString(R.string.default_web_client_id)
-                )
-                .requestEmail()
-                .build()
-    }
 
     @Test
-    fun clickOnSignInShouldLaunchIntent() {
+    fun clickOnButtonShouldSignInIfUserNull() {
         Intents.init()
-        onView(withId(R.id.signInButton)).check(matches(isClickable())).perform(click())
 
-        val mGoogleSignInClient = GoogleSignIn.getClient(
-                MainApplication.applicationContext(),
-                getGSO()
-        )
-        Intents.intended(IntentMatchers.filterEquals(mGoogleSignInClient.signInIntent))
+        onView(withId(R.id.signInButton))
+            .check(matches(isClickable())).perform(click())
+
+        //setting name to null makes currentUser return null and button will sign in
+        mockService.MOCK_NAME = "null"
+
+        val mockIntent = mockService.signInIntent()
+
+        IntentMatchers.filterEquals(mockIntent)
+
         Intents.release()
     }
 
+    /*@Test
+    fun clickOnButtonShouldSignOutIfUserPresent() {
+        Intents.init()
+
+        onView(withId(R.id.signInButton))
+                .check(matches(isClickable())).perform(click())
+
+        val mockIntent = mockService.signInIntent()
+
+        Intents.intended(IntentMatchers.filterEquals(mockIntent))
+
+        Intents.release()
+    }*/
+
```

Comment:
Is it normal that this test is commented out?

Reply:
This test should be removed. Thanks for pointing it out!